test/ketu/spec_test.clj

```diff
@@ -0,0 +1,134 @@
+                                                             (onCompletion [_ _ _])))
+                     create-callback           (gen/elements [boolean? int? contains? neg?])
+                     create-callback-obj       (gen/elements [boolean? int? contains? neg?])]
+        (let [sink-opts {:ketu/name                           name
+                         :ketu/topic                          topic
+                         :ketu.sink/shape                     shape
+                         :ketu.sink/sender-threads-num        sender-threads-num
+                         :ketu.sink/sender-threads-timeout-ms sender-threads-timeout-ms
+                         :ketu.sink/close-producer?           close-producer?
+                         :ketu.sink/producer-close-timeout-ms producer-close-timeout-ms
+                         :ketu.sink/callback                  callback
+                         :ketu.sink/callback-obj              callback-obj
+                         :ketu.sink/create-callback           create-callback
+                         :ketu.sink/create-callback-obj       create-callback-obj}]
+          (is (s/valid? :ketu/sink-opts sink-opts))))))
+
+  (testing "that passing valid required fields and invalid optional fields is invalid"
+    (tc/quick-check
+      50
+      (prop/for-all [name                      (gen/such-that (complement clojure.string/blank?) gen/string)
+                     topic                     gen/large-integer
+                     shape                     gen/string
+                     sender-threads-num        gen/string
+                     sender-threads-timeout-ms gen/string
+                     close-producer?           gen/string
+                     producer-close-timeout-ms gen/string
+                     callback                  gen/string
+                     callback-obj              gen/string
+                     create-callback           gen/string
+                     create-callback-obj       gen/string]
+        (let [sink-opts {:ketu/name                           name
+                         :ketu/topic                          topic
+                         :ketu.sink/shape                     shape
+                         :ketu.sink/sender-threads-num        sender-threads-num
+                         :ketu.sink/sender-threads-timeout-ms sender-threads-timeout-ms
+                         :ketu.sink/close-producer?           close-producer?
+                         :ketu.sink/producer-close-timeout-ms producer-close-timeout-ms
+                         :ketu.sink/callback                  callback
+                         :ketu.sink/callback-obj              callback-obj
+                         :ketu.sink/create-callback           create-callback
+                         :ketu.sink/create-callback-obj       create-callback-obj}]
+          (is (not (s/valid? :ketu/sink-opts sink-opts)))))))
+
+  (testing "that passing undocumented fields is valid"
```

Comment:
Nice one. This is actually checking that we are "open" and not "closed" for user custom data. Maybe change the `testing` string to reflect that.

Reply:
Done